src/loaders/lottie/tvgLottieBuilder.cpp

```diff
@@ -1202,9 +1195,21 @@ void LottieBuilder::updateMasks(LottieLayer* layer, float frameNo)
             //Cheaper. Replace the masking with a clipper
             if (layer->masks.count == 1 && compMethod == MaskMethod::Alpha) {
                 layer->scene->opacity(MULTIPLY(layer->scene->opacity(), opacity));
+                //Introduce an intermediate scene for embracing the precomp clipping + masking replaced by clipping
```

Comment:
Please check in 1177:
```
    if (layer->matteTarget || layer->type == LottieLayer::Precomp) {
        auto scene = Scene::gen();
        scene->push(layer->scene);
        layer->scene = scene;
    }
```

Reply:
the reasons why I didn't want to  combine these two are:
- matte + masking - an intermediate scene is _**not needed**_ if masking is replaced by clipping
- precomp + masking - an intermediate scene is _**needed**_ only if masking is replaced by clipping

but I understand that code is not so clear now, so I'll change this